build.assets/makefiles/master/k8s-master/cluster-dns.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Prior to upgrade to Kubernetes 1.19 the list of cluster DNS servers are provided
+# through the kubelet flag `--cluster-dns`. This flag is now deprecated and the cluster
+# DNS servers must be specified via the kubelet config file under `clusterDNS`.
+#
+# This script writes the cluster DNS servers into the kubelet config file.
+#
+# More info on `--cluster-dns` can be found at: https://v1-19.docs.kubernetes.io/docs/reference/command-line-tools-reference/kubelet/
+
+# Environment file that contains the cluster DNS_ADDRESSES.
+DNS_ENV=/run/dns.env
+# Path to kubelet config file.
+KUBELET_CONFIG=/etc/kubernetes/kubelet.yaml
+
+# If /run/dns.env file does not exist exit without any changes.
+if [ ! -f $DNS_ENV ]
+then
+    echo "$DNS_ENV does not exist"
+    exit
+fi
+
+# Import DNS_ADDRESSES from env file.
+export $(cat $DNS_ENV | xargs)
+
+# If DNS_ADDRESSES is empty exit without any changes.
+if [ -z "$DNS_ADDRESSES" ]
+then
+    echo "DNS_ADDRESSES is not set"
+    exit
+fi
+
+# Remove existing clusterDNS values.
+perl -i -p0e 's/clusterDNS:(\n  - .*)*\n//g' $KUBELET_CONFIG
+
+# Write new clusterDNS values to kubelet config.
+CLUSTER_DNS="clusterDNS:"
+while IFS=',' read -ra ADDR; do
+    for i in "${ADDR[@]}"; do
+        CLUSTER_DNS+="\n  - $i"
```

Comment:
The same here - could be achieved by using sed or tee right here and don't call `echo -e` later.

Reply:
What is the benefit of using `tee` over `echo`?

Reply:
I mean you can use sed/tee right here on line 40 without calling echo later

Reply:
I spent some time trying to figure out how to delete and append in the same command, but wasn't able to get it figured out. I don't want to spend too much time trying to learn how to use `sed` so I'll just leave it like this for now as long as it is producing the correct results.

Reply:
I think that's fine if it's working.